### ISSUE TITLE
High memory usage

### ISSUE THREAD

**Issue by tusquitoni** [pdulvp/jellyfin-qnap#95]
Hello,

I have installed this Jellyfin on my Qnap NAS with 32 GB RAM and I have a really bad RAM consumption problem. When Jellyfin is active, without doing any work, it consumes almost 50% of the RAM. But just by scanning a library I get even more consumption, reaching to saturate the NAS completely as it reaches 99% of RAM consumption. This causes the NAS to stop the process to protect itself.

I know that Jellyfin consumes a lot of RAM being a .Net application, but in my case I can not use it because it directly saturates the NAS and stops working. Does it happen to anyone else? Thank you.

Best regards.

**Comment by Oednerich** [comment]
Hey there,

can you tell me a bit more about your whole setup?
NAS Model, CPU, QTS Version, Jellyfin version?
do you use additional jellyfin plugins?
anything unusual in the jellyfin logfile?
the jellyfin logs are located in /share/CACHEDEV1_DATA/.qpkg/jellyfin/logs/ and should have a name like log_20240730.log for todays log for example.

I did some casual testing on my TS-664's RAM consumption, I'm also using 32GB.
There is not much else running on the system besides jellyfin right now.
tested on my TS-664 with Intel N5105, QTS 5.1.8.2823 build 20240712 and jellyfin 10.9.7 right here from pdulvp.

My idle RAM consumption from the jellyfin process is at 386,7 MB after letting the NAS and jellyfin run idle for around one hour.
With one 4k stream running form network it's at 557 MB and with 2 parallel 4k streams it's at 626,4 MB.

I think your RAM consumption is way too high.


![jellyfin-idle](https://github.com/user-attachments/assets/73614b31-ba66-4182-aed5-f87525a35dd3)
![jellyfin-1x4kstream](https://github.com/user-attachments/assets/a8a94fde-c0d5-46c2-925d-de4ab7f57aa2)
![jellyfin-2x4kstream](https://github.com/user-attachments/assets/516a7da0-afbe-42a3-9a99-b48aa11a1657)

**Comment by OneCDOnly** [comment]
> sorry for the big screenshots, is there a way on github to put something like spoiler tags around it?

Yes: https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab

**Comment by robotreto** [comment]
TS664 with 16GB, 16TB multimedia folder with >12000 MKV and MP4 files. Uses 1.5 GB RAM after startup, and 3.1 GB after a couple of hours.

**Comment by Oednerich** [comment]
> > sorry for the big screenshots, is there a way on github to put something like spoiler tags around it?
> 
> Yes: https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab



> > sorry for the big screenshots, is there a way on github to put something like spoiler tags around it?
> 
> Yes: https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab

thanks, thats actually very helpful!
But when using it on uploaded screenshots directly in the window it will look like this:
![summarytagsonuploadedimages](https://github.com/user-attachments/assets/2ad0a095-29ae-47e2-87cd-d790028e11f5)

**Comment by OneCDOnly** [comment]
> thanks, thats actually very helpful! But when using it on uploaded screenshots directly in the window it will look like this:
> my uploaded screenshots

Ah, I didn't know this. 🤓 

So, I messed with it a bit and it turns out you need a line space after the closing summary tag and before the image tag:
<details>
<summary>your uploaded screenshots</summary>

![jellyfin-idle](https://github.com/user-attachments/assets/73614b31-ba66-4182-aed5-f87525a35dd3) ![jellyfin-1x4kstream](https://github.com/user-attachments/assets/a8a94fde-c0d5-46c2-925d-de4ab7f57aa2) ![jellyfin-2x4kstream](https://github.com/user-attachments/assets/516a7da0-afbe-42a3-9a99-b48aa11a1657)
</details>

**Comment by Oednerich** [comment]
@OneCDOnly 

thanks! oh I feel a bit stupid right now, I thought about adding spaces but then thought "why would this matter" and didnt try.

**Comment by robotreto** [comment]
As I too have this ever increasing memory consumption on my qnap (growing from 550MB to 4GB in 2 weeks), I took the time to download the newest source (10.10.0) and compiled it under windows to analyze the memory consumption with the VS2022 debugger. During the initial library scan, RAM usage went up to 2.5GB. Subsequent re-scans of the libraries did not add to that. Then restarted the server, initial RAM usage was 400MB. First library rescan jumped to 600MB. All subsequent rescans (I tried more than 10) did not add any significant RAM usage, so it seemed stable. I also checked if the Garbage-Collector was called after library scans, and it was indeed called. So it seems indeed to be very hard to reproduce the problem. Maybe the memory-administration in windows is different than in Linux ?

**Comment by derpindialer** [comment]
How big are your libraries? I don't have high memory usage on my qnap jellyfin, but I do on my non qnap jellyfin. The difference between the two are my qnap jellyfin has a movie collection of about 300 and tv show collection of about 60 shows. On the non qnap jellyfin, I have about 8 large libraries, some having around 3k movies, and it seems to use a huge amount of memory during scanning. When I lowered the size of the libraries, the memory usage was lower.

**Comment by robotreto** [comment]
14'000 MKV's using 17TB, plus about 5'000 mp3 using 0.05TB. Memory usage after restart is 550MB, after 3 weeks 6GB.
I don't think this memory leak is specific for the QNAP port, it seems to happen also on other linux ports including containers. I once tried windows jellyfin on my large library, there I could not see this insane memory growth.

**Comment by nickb937** [comment]
My Jellyfin instance on Linux (not QNAP) is 2.5GB RSS.  This isn't a problem specific to the QNAP port, it is just related to the way Jellyfin utilises memory for libraries
